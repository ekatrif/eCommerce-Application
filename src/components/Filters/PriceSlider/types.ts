export interface IPrice {
  price: {
    min: number;
    max: number;
  };
  value: number[];
  changePrice: () => void;
}
