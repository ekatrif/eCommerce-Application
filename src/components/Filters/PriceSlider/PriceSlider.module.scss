.slider {
  margin: 2em auto 1em;
  width: 80%;
}