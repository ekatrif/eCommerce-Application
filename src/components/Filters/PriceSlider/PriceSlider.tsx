import * as React from 'react';
import Slider from '@mui/material/Slider';
import classes from './PriceSlider.module.scss';
import { IPrice } from './types';

const PriceSlider: React.FC<IPrice> = ({ price }: IPrice) => {
  const [value, setValue] = React.useState<number>(price.min);

  const handleChange = (event: Event, newValue: number | number[]): void => {
    setValue(newValue as number);
  };

  return (
    <div className={classes.slider}>
      <Slider
        aria-label="Price"
        defaultValue={price.min}
        value={value}
        onChange={handleChange}
        valueLabelDisplay="auto"
        sx={{
          color: '#247C52',
        }}
        min={price.min}
        max={price.max}
      />
    </div>
  );
};

export default PriceSlider;
