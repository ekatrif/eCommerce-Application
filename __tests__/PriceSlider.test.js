import React from 'react';
import renderer from 'react-test-renderer';
import PriceSlider from '../src/components/Filters/PriceSlider/PriceSlider';

test('PriceSlider renders correctly', () => {
  const tree = renderer
    .create(<PriceSlider price={{
      min: 0,
      max: 100,
    }}
    value="10"
    changePrice="" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});