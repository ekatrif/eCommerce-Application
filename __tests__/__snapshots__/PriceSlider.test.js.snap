// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriceSlider renders correctly 1`] = `
<div
  className="slider"
>
  <span
    className="MuiSlider-root MuiSlider-colorPrimary MuiSlider-sizeMedium css-u259bk-MuiSlider-root"
    onMouseDown={[Function]}
  >
    <span
      className="MuiSlider-rail css-14pt78w-MuiSlider-rail"
    />
    <span
      className="MuiSlider-track css-1gv0vcd-MuiSlider-track"
      style={
        {
          "left": "0%",
          "width": "10%",
        }
      }
    />
    <span
      className="MuiSlider-thumb MuiSlider-thumbSizeMedium MuiSlider-thumbColorPrimary MuiSlider-thumb MuiSlider-thumbSizeMedium MuiSlider-thumbColorPrimary css-eg0mwd-MuiSlider-thumb"
      data-index={0}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      style={
        {
          "left": "10%",
          "pointerEvents": undefined,
        }
      }
    >
      <input
        aria-orientation="horizontal"
        aria-valuemax={100}
        aria-valuemin={0}
        aria-valuenow={10}
        data-index={0}
        disabled={false}
        max={100}
        min={0}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        step={1}
        style={
          {
            "border": 0,
            "clip": "rect(0 0 0 0)",
            "direction": "ltr",
            "height": "100%",
            "margin": -1,
            "overflow": "hidden",
            "padding": 0,
            "position": "absolute",
            "whiteSpace": "nowrap",
            "width": "100%",
          }
        }
        type="range"
        value={10}
      />
      <span
        aria-hidden={true}
        className="MuiSlider-valueLabelOpen MuiSlider-valueLabel css-nnid7-MuiSlider-valueLabel"
      >
        <span
          className="MuiSlider-valueLabelCircle"
        >
          <span
            className="MuiSlider-valueLabelLabel"
          >
            10
          </span>
        </span>
      </span>
    </span>
  </span>
</div>
`;
